.github/workflows/push.yml

```
@@ -0,0 +1,81 @@
+name: Push
+on:
+  push:
+    paths-ignore:
+      - docs/*
+      - NEWS
+      - UPGRADING
+      - UPGRADING.INTERNALS
+      - README.md
+      - CONTRIBUTING.md
+      - CODING_STANDARDS.md
+    branches:
+      - PHP-7.4
+      - PHP-8.0
+      - PHP-8.1
+      - master
+  pull_request:
+    branches:
+      - '**'
```

```suggestion
on:
  pull_request:
  push:
  schedule:
    - cron: '0 0/6 * * *'
```

testing should trigger for all pushes, expecially usefull for forks /wo PR created yet

also cron (for default branch) is usefull to retest periodically to discover external issues

---

You can also create PRs in your fork. Nightly needs more work.

---

> You can also create PRs

yes, I can, but why not test all pushes by a default? Thanks to GitHub, it is free :)

---

It will waste resources and burn the planet

---

I do not belive so, as the push CI will be executed in fork, thus it will not prolong the php-src queue in any way.

---

I think we should have something working first, and then we can still fine tune/improve. As it is for quite a while now, we only have AppVeyor CI for PHP-8.0 (i.e. we're only testing on Windows, which is rather suboptimal).

---

yes, Windows testing in GH Actions would be nice, especially in forks

@iluuu1994 please reconsider the push trigger, we might burn the planet a little bit, but for developer, at least for me, it brings value

---

What I'm saying is that we currently have *no* working *nix CI for the PHP-8.0 branch, what is *bad*. Having some, even if not optimal, is important. So please, don't let perfect be the enemy of good. :)

---

Creating a PR in your fork is trivial, so this solution allows both running and not running the CI when appropriate. So I'll keep it this way for now.